### DISCUSSION TITLE
qiita-cli にて pull を行った時に public/ 以下に .md ファイルのみが作成されるのではなく記事ごとのディレクトリが作成され、作成されたディレクトリ以下に .mdファイルが作成されるようになってほしい

### DISCUSSION THREAD

**Discussion by TakuKobayashi** [increments/qiita-discussions#515] [要望・アイデア・不具合]
現在、[qiita-cli](https://github.com/increments/qiita-cli) にて `qiita pull` をおこなった場合、 `指定したrootディレクトリ/public/qiitaに公開している記事のid.md` といった形で記事をダウンロードしております。これを `指定したrootディレクトリ/public/{qiitaに公開している記事のid(あるいは記事ごとに一意に識別できる文字列)}/qiitaに公開している記事のid.md` の形にしてダウンロードできるようにしてほしいと思いました。あるいはそのように指定できるオプションを追加してほしいと思いました。
このようにしてほしい理由として、今後、記事内で使用している画像を[qiita-cli](https://github.com/increments/qiita-cli) 経由でアップロードすることができるようになった場合、記事である `.md` ファイル内で参照している画像はローカルで使用している画像を参照するようになります。
つまり 1つの記事を構成するファイルは `.md` ファイルのみならず、その記事内で参照している画像ファイル一式も記事を構成する要素となります。そのため記事ごとに `指定したrootディレクトリ/public/{qiitaに公開している記事のid(あるいは記事ごとに一意に識別できる文字列)}/qiitaに公開している記事のid.md` のような形でディレクトリが区切られていれば記事内で使用している画像など一式をローカルで管理し、そのまま記事として投稿できるようになります。
そのため、[qiita-cli](https://github.com/increments/qiita-cli)においてこのような `qiita pull` を行う場合、`指定したrootディレクトリ/public/{qiitaに公開している記事のid(あるいは記事ごとに一意に識別できる文字列)}/qiitaに公開している記事のid.md` のような階層構造の形で記事をダウンロードできれば記事内で使用している画像の管理も行いやすくなるので、ぜひこのような構造でのダウンロードが可能なように [qiita-cli](https://github.com/increments/qiita-cli) の機能(仕様)として取り入れていただけるといいなと思います。よろしくお願いします。

**Comment by ohakutsu** [comment]
@TakuKobayashi 
ご意見ありがとうございます！

Qiita CLIの改善・アップデートについて考えていただけて嬉しいです！

今後、Qiita CLIで画像を管理できるようになった際に、以下のように記事単位で画像を管理できるようになって欲しいというふうに解釈させていただきました。

```
└── public/
   ├── 記事ディレクトリ1/
   │  ├── 記事ファイル1.md
   │  ├── 画像ファイルA
   │  └── 画像ファイルB
   └── 記事ディレクトリ2/
      ├── 記事ファイル2.md
      └── 画像ファイルC
```

画像ファイルの配置場所に関して、以下のように`images`ディレクトリに集約させる配置なども方法としてはあると思っており、どの配置にするかまだ結論を出せていないのが現状となっていました。

```
├── images/
│  ├── 画像ファイルA
│  └── 画像ファイルB
└── public/
   ├── 記事ファイル1
   └── 記事ファイル2
```

そのため、ご意見いただいたディレクトリ構造にすることをお約束できませんが、貴重なご意見としてアップデートに役立たせていただきます

**Reply by TakuKobayashi** [reply]
@ohakutsu 
お返事ありがとうございます!!
はい、今回のご要望は例としてご提示いただきました

```
└── public/
   ├── 記事ディレクトリ1/
   │  ├── 記事ファイル1.md
   │  ├── 画像ファイルA
   │  └── 画像ファイルB
   └── 記事ディレクトリ2/
      ├── 記事ファイル2.md
      └── 画像ファイルC
 ```
 
こちらのディレクトリ構造になることをご要望といたしました。
私個人の意見で恐縮ですが上記のように記事ごとにディレクトリを分けた管理方法の方が `images` 以下に画像を保存する管理方法よりも管理がしやすかったのでできたらこちらの方がいいなぁと思い、ご要望いたしました。(両方の管理方法を試したことがあります。その時に `images` 以下に画像を保存する管理方法だと記事の数が多くなるとどの記事でどの画像を参照したらいいかわからなくなってしまいました...)
いずれの形にせよご検討いただき、記事のディレクトリの構造を変更した形でのアップデートがなされると幸いです。
よろしくお願いします

**Reply by ohakutsu** [reply]
@TakuKobayashi 
ありがとうございます、確かにimages/の形だと画像の対応がわかりにくいデメリットがありそうですね...
運営としても使いやすさ、実現しやすさを鑑みて検討をしていこうと思います。
またQiita CLIを使っている中で何かあればご意見くださると嬉しいです！
